cmd/build.go

```diff
@@ -13,36 +13,43 @@ import (
-	buildCmd.Flags().StringP("image", "i", "", "Optional full image name, in form [registry]/[namespace]/[name]:[tag] for example quay.io/myrepo/project.name:latest (overrides --registry) - $FUNC_IMAGE")
-	buildCmd.Flags().StringP("path", "p", cwd(), "Path to the Function project directory - $FUNC_PATH")
-	buildCmd.Flags().StringP("registry", "r", "", "Registry for built images, ex 'docker.io/myuser' or just 'myuser'.  Optional if --image provided. - $FUNC_REGISTRY")
+	buildCmd.Flags().StringP("builder", "b", "default", "Buildpack builder, either an as a an image name or a mapping name as defined in func.yaml")
+	buildCmd.Flags().BoolP("confirm", "c", false, "Prompt to confirm all configuration options (Env: $FUNC_CONFIRM)")
+	buildCmd.Flags().StringP("image", "i", "", "Full image name in the orm [registry]/[namespace]/[name]:[tag] (optional). This option takes precedence over -registry (Env: $FUNC_IMAGE")
+	buildCmd.Flags().StringP("path", "p", cwd(), "Path to the project directory (Env: $FUNC_PATH)")
+	buildCmd.Flags().StringP("registry", "r", "", "Registry + namespace part of the image to build, ex 'quay.io/myuser'.  The full image name is automatically determined based on the local directory name. If not provided the registry will be taken from func.yaml (Env: $FUNC_REGISTRY)")
 
 	err := buildCmd.RegisterFlagCompletionFunc("builder", CompleteBuilderList)
 	if err != nil {
-		fmt.Println("Error while calling RegisterFlagCompletionFunc: ", err)
+		fmt.Println("internal: error while calling RegisterFlagCompletionFunc: ", err)
 	}
 }
 
 var buildCmd = &cobra.Command{
 	Use:   "build",
-	Short: "Build an existing Function project as an OCI image",
-	Long: `Build an existing Function project as an OCI image
-
-Builds the Function project in the current directory or in the directory
-specified by the --path flag. The func.yaml file is read to determine the
-image name and registry. If both of these values are unset in the
-configuration file the --registry or -r flag should be provided and an image
-name will be derived from the project name.
-
-Any value provided for the --image flag will be persisted in the func.yaml
-configuration file. On subsequent invocations of the "build" command
-these values will be read from the configuration file.
-
-It's possible to use a custom Buildpack builder with the --builder flag.
-The value may be image name e.g. "cnbs/sample-builder:bionic",
-or reference to builderMaps in the config file e.g. "default".
+	Short: "Build a function project as a container image",
+	Long: `Build a function project as a container image
+
+This command builds the function project in the current directory or in the directory
+specified by --path. The result will be a container image that is pushed to a registry.
+The func.yaml file is read to determine the image name and registry. 
+If func.yaml does not exists, either --registry or --image has to be provided and the 
+configuration file be created automatically.
```

Comment:
```suggestion
If project has not been previously built, either --registry or --image must be provided and the 
image name will be written to the configuration file.
```

Reply:
I already phrased it a little bit differently:

```
If the project has not already been built, either --registry or --image must be provided 
and the image name is stored in the configuration file.
```